### ISSUE TITLE
TypeError: countries.map is not a function

### ISSUE THREAD

**Issue by aprendiz20** [adrianhajdin/project_corona_tracker#4]
First of all let me thank you,  for sharing  this valuable knowledge
I coded along with  you and everything  went well for the first couple of  minutes. However in the end I´m getting the following error: 

   TypeError: countries.map is not a function
   19 | return (
  20 |     <FormControl className={styles.formControl}>
  21 |         <NativeSelect defaultValue="" onChange={(e) => handleCountrychange(e.target.value)}>
> 22 |             <option value="">Global</option>
     | ^  23 |             {countries.map((country, i) => <option key={i} value={country}>{country}</option>)}
  24 |         </NativeSelect>
  25 |     </FormControl>

**Comment by kunalkashyap855** [comment]
The name of the variable here should be what you have written in useState. According to the video, the name of the state variable is 'fetchedCountries' instead of 'countries', that might be the problem.

**Comment by kunalkashyap855** [comment]
Also in your line 23, you're closing the function with curly braces instead of parentheses.

**Comment by FottyM** [comment]
Can you share the entire code snippet?

**Comment by harshilparmar** [comment]
@FottyM Brother @kunalkashyap855  is saying that you should write like this
`{fetchedCountries.map((country, i) => {country})}`

**Comment by aprendiz20** [comment]
I managed  fix the problem  with  your support. Thank you!

**Comment by gellys** [comment]
Hello guys, taking advantage of the fact that it's still open, I'm having a similar problem:

TypeError: fetchedCountries.map is not a function
CountryPicker
tracker/src/components/CountryPicker/CountryPicker.jsx:21
  18 | return (
  19 |   <FormControl className={styles.formControl}>
  20 |     <NativeSelect>
> 21 |       <option value="global">Global</option>
     | ^  22 |       {fetchedCountries.map((country, i) => (
  23 |         <option key={i} value={country}>
  24 |           {country}

Below is how my code is :

![question-function](https://user-images.githubusercontent.com/27901470/83689842-1acd6f00-a5c6-11ea-8f8f-2e1191e12bb4.PNG)

**Comment by FottyM** [comment]
@gellys `setSate(setFetchedCountries)` is expecting an array but I am getting what it is getting is an object hence the error. 
What you can do is
```javascript
const fetchApi = async () => {
const countries = await (await fetch('put URL here')).json()
setFetchedCountries(countries)
}

// await (await fetch('put URL here')).json() is the as
// -----
// const response  = await fetch(url)
// const countries = await response.json()
```
the issue is you can't pass `await fetchCountries` to `setFetchedCountries`

**Comment by harshilparmar** [comment]
@FottyM  yes you are right, @gellys  `setFetchedCountries(await fetchCountries)` is not calling method,so I think it is getting whole reference of fetchCountries method.

**Comment by gellys** [comment]
> Edit:
> @gellys `setFetchedCountries` is expecting an array but I am guessing what it is getting is an object because you are passing `await fetchCountries` to `setFetchedCountries` what you could do is.
> 
> ```ts
> const countries = await fetchCountries()
> setFetchedCountries(countries)
> 
> // or 
> setFetchedCountries(await fetchCountries())
> ```

That's it! Now it worked... Thanks @FottyM :100:  :+1:

**Comment by sagar-maheshwari653** [comment]
I am still getting the same error: " TypeError: Cannot read property 'map' of undefined " and also getting the error: " TypeError: Cannot read property 'confirmed' of undefined ", code is provided below: 

![Screenshot (16)](https://user-images.githubusercontent.com/72984307/123085520-f5650a80-d43f-11eb-9dd2-13db2dc23473.png)
![Screenshot (17)](https://user-images.githubusercontent.com/72984307/123085529-f72ece00-d43f-11eb-8f75-a89a7ad23193.png)

![Screenshot (19)](https://user-images.githubusercontent.com/72984307/123085701-2f361100-d440-11eb-9e17-4bc066f35480.png)
![Screenshot (22)](https://user-images.githubusercontent.com/72984307/123085843-568cde00-d440-11eb-9dd3-a03dc8c7c1cb.png)
